### ISSUE TITLE
TF Binding NaN loss issue

### ISSUE THREAD

**Issue by chanil1218** [baidu-research/warp-ctc#64]
When I switched from `tf.nn.ctc_loss` to `warpctc_tensorflow.ctc` I got NaN loss for the early moment of training even though `tf.nn.ctc_loss` could learn normally. 

![warp-ctc-nan](https://cloud.githubusercontent.com/assets/1420133/21585002/9ffbb17c-d0fa-11e6-860d-492d287f5605.png)

I wouldn't know how should I make this issue reproducible on minimal configuration, I'm training https://github.com/buriburisuri/speech-to-text-wavenet this implementation with VCTK corpus only changing `tf.nn.ctc_loss` to `warpctc_tensorlflow.ctc`. 

I had experienced NaN loss issue frequently(not always) on torch binding, and at that time I thought this NaN issue comes from training set. But when I could switched between tf implementation of ctc loss and warp-ctc, I am now suspect this NaN loss issue originated from warp-ctc core. 

In the experience from torch binding of warp-ctc, warp-ctc would return NaN, inf, or -inf loss often.

**Comment by chanil1218** [comment]
I'm using Python 3.5.2 with Tensorflow 0.12.0 compiled from source code. 
I've tested ctc_op_kernel version, and I confirmed that it did work normally.

**Comment by jaredcasper** [comment]
Are you accounting for the fact that the "reserved" symbol defaults to `0` for the `warpctc_tensorflow.ctc` op and the highest symbol value for the `tf.nn.ctc_loss` op?  To switch between the two ops you need to either make sure your labels don't use the reserved symbol or set the `blank_label` parameter in `warpctc_tensorflow.ctc`.

The inputs are of course in different formats as well... are you sure you are sending in the label lengths, etc. correctly?  Also check that when using `tf.nn.ctc_loss` that `proprocess_collapse_repeated` is `False` and `ctc_merge_repeated` is `True` (their default values).  `warp_ctc` does not support the other configurations that the TF ctc does.

The torch bindings use the exactly same kernels under the hood, the only difference is shuttling the inputs into the functions.  So if it's working better for you in Torch I would suggest looking closely at the inputs to the warp_ctc kernel in the TF version to make sure they are what you want them to be.  If you can get inputs that produce a NaN or +/-inf in the TF op but not in the Torch op that would indeed be interesting.
